spec-tests/src/test/java/org/candlepin/spec/content/OwnerContentSpecTest.java

```diff
@@ -1157,4 +958,1038 @@ public void shouldPermitSafeParallelModification() throws Exception {
+        }
+
+        @Test
+        public void shouldDefaultToActiveExclusiveCustomInclusive() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedCids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(ProductInfo::active)
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<ContentDTO> output = this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), null, null, null, null);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ContentDTO::getId)
+                .containsAll(expectedCids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @Test
+        public void shouldFailWithInvalidOwners() {
+            assertNotFound(() -> this.adminClient.ownerContent()
+                .getContentsByOwner("invalid_owner", null, null, null, null));
+        }
+
+        @Test
+        public void shouldAllowFilteringOnIDs() {
+            // Randomly seeded for consistency; seed itself chosen randomly
+            Random rand = new Random(58258);
+
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> pids = this.productMap.values()
+                .stream()
+                .filter(elem -> rand.nextBoolean())
+                .map(ProductInfo::id)
+                .toList();
+
+            List<String> expectedCids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(prod -> pids.contains(prod.id()))
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<ContentDTO> output = this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), expectedCids, null, INCLUSION_INCLUDE, INCLUSION_INCLUDE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ContentDTO::getId)
+                .containsAll(expectedCids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = { "invalid_id" })
+        @NullAndEmptySource
+        public void shouldAllowFilteringOnInvalidIds(String cid) {
+            // This test verifies that invalid IDs are "allowed", but they won't match on anything
+
+            OwnerDTO owner = this.owners.get(1);
+
+            ProductDTO expected = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .findAny()
+                .map(ProductInfo::dto)
+                .get();
+
+            List<String> expectedPids = List.of(expected.getId());
```

Comment:
Same here. We're still being very product centric for content-specific tests.

Reply:
updated